ui/src/containers/SampleFlexView.jsx

```diff
@@ -1,156 +1,245 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
 import { Col } from 'react-bootstrap';
-import { connect } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 
 import { filterAction } from '../actions/sampleGrid';
 
-class NewSampleFlexView extends React.Component {
-  constructor(props) {
-    super(props);
+export default function SampleFlexView(props) {
+  const {
+    displayPuckCellContextMenu,
+    cellMenuID,
+    puckMenuID,
+    type = 'FLEX HCD',
+  } = props;
 
-    this.renderCircle = this.renderCircle.bind(this);
-    this.isCellSelected = this.isCellSelected.bind(this);
+  const filterOptions = useSelector((state) => state.sampleGrid.filterOptions);
+  const scContent = useSelector((state) => state.sampleChanger.contents);
+  const [pendingEvent, setPendingEvent] = useState(null);
+
+  const numberOfCell = scContent.children.length; // Number of cells/slices
+
+  const dispatch = useDispatch();
+
+  function handleClickonCellPuck(event, cellID, puckID) {
+    dispatch(
+      filterAction({ cellFilter: `${cellID}`, puckFilter: `${puckID}` }),
+    );
+    event.stopPropagation();
   }
 
-  componentDidMount() {
-    document.addEventListener('keydown', this.onKeyDown, false);
+  function isCellSelected(cellID) {
+    return Number(filterOptions.cellFilter) === cellID;
   }
 
-  onClickCell(event, idx, disabled) {
-    if (!disabled) {
-      this.props.filter({ cellFilter: `${idx + 1}` });
+  function isPuckSelected(cellID, puckID) {
+    return (
+      Number(filterOptions.cellFilter) === cellID &&
+      Number(filterOptions.puckFilter) === puckID
+    );
+  }
+
+  // Call displayPuckCellContextMenu only after Redux state updates
```

Comment:
This feels very hacky. You can call something after a Redux state update by waiting for the promise returned by a dispatch call to resolve:

```js
await dispath(...);
displayPuckCellContextMenu(...);
```

Saves you from needing that `pendingEvent` local state.